packages/messaging/type.ts

```diff
@@ -0,0 +1,11 @@
+export interface PayloadAndResponse<P = unknown, D = unknown> {
+  payload: P;
+  response: D;
+}
+/**
+ * Define the type of messages
+ * If you want to add a new message type, you can add it here.
+ */
+export interface Messages {
+  foo: PayloadAndResponse<{ bar: string }, string>;
```

Comment:
Maybe more meaningful naming, it's pretty hard to understand at first with foo: bar, what is the protocol what is request types and return types. Maybe HandshakeProtocol, RequestHandshake, ResponseHandshake etc.?